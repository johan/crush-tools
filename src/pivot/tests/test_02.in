Name	Blank-0	Blank-1	ID	Met-0	Met-1	Met-2	Met-3 	Met-4	Met-5	Met-6	Met-7	Met-8	Met-9	Met-10  Met-11  Met-12  Met-13  Met-14  Met-15  Data
Cheese			0762	35	14	1.87	0.13	40.00	1.10	3	21.43	0.62	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			0831	81	15	7.34	0.49	18.52	1.50	3	20.00	2.45	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			0860	14	4	1.22	0.30	28.57	1.10	1	25.00	1.22	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			0926	1393	67	76.38	1.14	4.81	2.50	4	5.97	19.09	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			0944	29	2	1.63	0.81	6.90	2.40	1	50.00	1.63	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			0955	114	2	1.43	0.71	1.75	6.50	1	50.00	1.43	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			0958	3660	214	17.32	0.08	5.85	1.10	17	7.94	1.02	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			0989	153	22	3.41	0.15	14.38	1.40	4	18.18	0.85	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1044	178	73	13.87	0.19	41.01	1.40	6	8.22	2.31	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1080	150	30	2.50	0.08	20.00	2.00	1	3.33	2.50	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1122	94	10	1.68	0.17	10.64	1.20	3	30.00	0.56	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1164	30	8	5.48	0.69	26.67	1.20	1	12.50	5.48	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1198	46	6	1.35	0.23	13.04	1.50	1	16.67	1.35	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1202	360	47	16.98	0.36	13.06	1.70	3	6.38	5.66	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1213	13	3	0.09	0.03	23.08	1.00	1	33.33	0.09	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1222	211	56	2.54	0.05	26.54	1.10	10	17.86	0.25	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1224	12	3	1.28	0.43	25.00	1.70	1	33.33	1.28	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1266	100	18	1.53	0.09	18.00	1.30	3	16.67	0.51	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1297	112	19	2.38	0.13	16.96	1.50	2	10.53	1.19	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1331	48	3	1.90	0.63	6.25	1.70	1	33.33	1.90	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1332	773	226	17.73	0.08	29.24	1.20	40	17.70	0.44	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1349	18	3	0.22	0.07	16.67	1.20	1	33.33	0.22	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1374	1778	27	28.67	1.06	1.52	2.00	1	3.70	28.67	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1378	446	40	16.39	0.41	8.97	1.80	1	2.50	16.39	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1450	29	3	2.22	0.74	10.34	1.70	3	100.00	0.74	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1475	42	7	1.73	0.25	16.67	1.00	1	14.29	1.73	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1482	73	17	1.60	0.09	23.29	1.00	2	11.76	0.80	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1516	110	0	0.00	0.00	0.00	4.60	1	0.00	0.00	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1588	17	8	1.58	0.20	47.06	1.30	1	12.50	1.58	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1636	24	6	0.65	0.11	25.00	1.90	1	16.67	0.65	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1649	79	7	0.72	0.10	8.86	1.40	6	85.71	0.12	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1658	92	1	0.92	0.92	1.09	1.70	1	100.00	0.92	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			1659	128	34	2.76	0.08	26.56	1.10	1	2.94	2.76	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			2305	69	2	1.85	0.93	2.90	2.60	1	50.00	1.85	0	0.00	0.00	0.00	0.00	0.00		stringy
Cheese			2334	68	1	0.55	0.55	1.47	2.10	1	100.00	0.55	0	0.00	0.00	0.00	0.00	0.00		stringy
