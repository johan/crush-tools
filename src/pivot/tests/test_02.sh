test_number=02
description="segfault bug fix"

input=$test_dir/test_$test_number.in
expected=$test_dir/test_$test_number.expected

subtest=1
output=$test_dir/test_$test_number.$subtest.output
$bin -k -f 1-10 -p 20,21 -v 11,12,18,19 $input > $output
if [ $? -ne 0 ] || [ "`diff -q $output $expected`" ]; then
  test_status $test_number $subtest "$description (indexes)" FAIL
else
  test_status $test_number $subtest "$description (indexes)" PASS
  rm $output
fi

subtest=2
output=$test_dir/test_$test_number.$subtest.output
$bin -F Name,Blank-0,Blank-1,ID,Met-0,Met-1,Met-2,Met-3,Met-4,Met-5 \
     -P Met-15,Data \
     -A Met-6,Met-7,Met-13,Met-14 $input > $output
if [ $? -ne 0 ] || [ "`diff -q $output $expected`" ]; then
  test_status $test_number $subtest "$description (labels)" FAIL
else
  test_status $test_number $subtest "$description (labels)" PASS
  rm $output
fi
