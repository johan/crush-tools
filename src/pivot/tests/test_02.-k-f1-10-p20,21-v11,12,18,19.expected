Name	Blank-0	Blank-1	ID	Met-0	Met-1	Met-2	Met-3	Met-4	Met-5	 - stringy: Met-6	 - stringy: Met-7	 - stringy: Met-13	 - stringy: Met-14
Cheese			0762	35	14	1.87	0.13	40.00	1.10	3	21.43	0.00	0.00
Cheese			0831	81	15	7.34	0.49	18.52	1.50	3	20.00	0.00	0.00
Cheese			0860	14	4	1.22	0.30	28.57	1.10	1	25.00	0.00	0.00
Cheese			0926	1393	67	76.38	1.14	4.81	2.50	4	5.97	0.00	0.00
Cheese			0944	29	2	1.63	0.81	6.90	2.40	1	50.00	0.00	0.00
Cheese			0955	114	2	1.43	0.71	1.75	6.50	1	50.00	0.00	0.00
Cheese			0958	3660	214	17.32	0.08	5.85	1.10	17	7.94	0.00	0.00
Cheese			0989	153	22	3.41	0.15	14.38	1.40	4	18.18	0.00	0.00
Cheese			1044	178	73	13.87	0.19	41.01	1.40	6	8.22	0.00	0.00
Cheese			1080	150	30	2.50	0.08	20.00	2.00	1	3.33	0.00	0.00
Cheese			1122	94	10	1.68	0.17	10.64	1.20	3	30.00	0.00	0.00
Cheese			1164	30	8	5.48	0.69	26.67	1.20	1	12.50	0.00	0.00
Cheese			1198	46	6	1.35	0.23	13.04	1.50	1	16.67	0.00	0.00
Cheese			1202	360	47	16.98	0.36	13.06	1.70	3	6.38	0.00	0.00
Cheese			1213	13	3	0.09	0.03	23.08	1.00	1	33.33	0.00	0.00
Cheese			1222	211	56	2.54	0.05	26.54	1.10	10	17.86	0.00	0.00
Cheese			1224	12	3	1.28	0.43	25.00	1.70	1	33.33	0.00	0.00
Cheese			1266	100	18	1.53	0.09	18.00	1.30	3	16.67	0.00	0.00
Cheese			1297	112	19	2.38	0.13	16.96	1.50	2	10.53	0.00	0.00
Cheese			1331	48	3	1.90	0.63	6.25	1.70	1	33.33	0.00	0.00
Cheese			1332	773	226	17.73	0.08	29.24	1.20	40	17.70	0.00	0.00
Cheese			1349	18	3	0.22	0.07	16.67	1.20	1	33.33	0.00	0.00
Cheese			1374	1778	27	28.67	1.06	1.52	2.00	1	3.70	0.00	0.00
Cheese			1378	446	40	16.39	0.41	8.97	1.80	1	2.50	0.00	0.00
Cheese			1450	29	3	2.22	0.74	10.34	1.70	3	100.00	0.00	0.00
Cheese			1475	42	7	1.73	0.25	16.67	1.00	1	14.29	0.00	0.00
Cheese			1482	73	17	1.60	0.09	23.29	1.00	2	11.76	0.00	0.00
Cheese			1516	110	0	0.00	0.00	0.00	4.60	1	0.00	0.00	0.00
Cheese			1588	17	8	1.58	0.20	47.06	1.30	1	12.50	0.00	0.00
Cheese			1636	24	6	0.65	0.11	25.00	1.90	1	16.67	0.00	0.00
Cheese			1649	79	7	0.72	0.10	8.86	1.40	6	85.71	0.00	0.00
Cheese			1658	92	1	0.92	0.92	1.09	1.70	1	100.00	0.00	0.00
Cheese			1659	128	34	2.76	0.08	26.56	1.10	1	2.94	0.00	0.00
Cheese			2305	69	2	1.85	0.93	2.90	2.60	1	50.00	0.00	0.00
Cheese			2334	68	1	0.55	0.55	1.47	2.10	1	100.00	0.00	0.00
